### ISSUE TITLE
DOC. typo in section of DataLoader

### ISSUE THREAD

**Issue by nuclear718** [FluxML/Flux.jl#1112]
Original
> Takes as input one or more data tensors, e.g. X in unsupervised learning, X and Y in supervised learning. The last dimension in each tensor is considered to be the observation dimension.

`Xtrain = rand(10, 100)`
`Ytrain = rand(100)`
`train_loader = DataLoader(Xtrain, Ytrain, batchsize=2, shuffle=true) `
`for epoch in 1:100`
`   for (x, y) in train_loader`
`       @assert size(x) == (10, 2)`
`        @assert size(y) == (2,)`
`        ...`
`    end`
`end`

The "last dimension" in this case is "100" right? 
rand(10,100)=> 10X100 matrix with random variable elements and the last dimension of the matrix is 100. The observation dimension should be "10" and the last dimension "100" should be the number of observations (or the number of instances).

The number of observations(instances) = the number of labels ( length(Ytrain )=100 )

Suggest to revise as:
> Takes as input one or more data tensors, e.g. X in unsupervised learning, X and Y in supervised learning. The last dimension in each tensor is considered to be the observation **number.**

**Comment by CarloLucibello** [comment]
observation dimension means the dimension along which the different observations are indexed. I agree it is slightly ambiguous, but I cannot think of anything better.

Observation number is not very meaningful, confusing dimensions and numbers)

**Comment by johnnychen94** [comment]
I believe the term "observation dimension" is more formally used than "batch dimension", "the last dimension", or "the N dimension".

Some references I can think of to this usage:

* [MLDataPattern](https://mldatapatternjl.readthedocs.io/en/latest/documentation/container.html#observation-dimension)
* [Matlab](https://www.mathworks.com/help/deeplearning/ref/checklayer.html)

**Comment by nuclear718** [comment]
> observation dimension means the dimension along which the different observations are indexed. I agree it is slightly ambiguous, but I cannot think of anything better.
> 
> Observation number is not very meaningful, confusing dimensions and numbers)

Yes, you are right! In linear algebra view point, we always say the dimension of observation matrix is 10X100. The "last dimension" is 100, that is correctly! 
But
let's look the standard [IRIS dataset ](https://gist.github.com/curran/a08a1080b88344b0c8a7)!
What is the dimension of this dataset, I will say dimension=4 **not dimension=150.**
Based on the datascience's view point, we care the feature space and it is build by "features".
Another dimension 150 is a kind of matrix dimension but in datascience the 150 is stand for number of data points not dimension in feature space.

**Comment by johnnychen94** [comment]
@nuclear718 I think you misread the sentence, in `Xtrain = rand(10, 100)`, it says the observation dimension is `2`. If we're speaking about the number of batches `100`, a more accurate description can be "the number of samples along with the observation dimension" or "the length of the observation dimension".
